generate-osm/map-db/db-mapper.go

```diff
@@ -33,10 +43,292 @@ func MapDB([]string refs, string osmDir, string DBDir) {
+	for _, signal := range knoten.HauptsigF {
+		found := false
+		kilometrage := signal.KnotenTyp.Kilometrierung[0].Value
+		if node_list := anchors[kilometrage]; node_list != nil {
+			for _, node := range node_list {
+				typ, err1 := OSMUtil.GetTag(*node, "type")
+				subtyp, err2 := OSMUtil.GetTag(*node, "subtype")
+				id, err3 := OSMUtil.GetTag(*node, "id")
+				direction, err4 := OSMUtil.GetTag(*node, "direction")
+				if err1 == nil && typ == "element" && err2 == nil && subtyp == "ms" && err3 == nil && id == signal.Name[0].Value && err4 == nil && direction == "falling" {
+					found = true
+					break
+				}
+			}
+		}		
+
+		if found {
+			continue
+		}
+
+		for _, node := range osmData.Node {
+			if len(node.Tag) == 0 {
+				continue
+			}
+
+			is_signal := false 
+			has_correct_id := false
+			if railwayTag, err := OSMUtil.GetTag(*node, "railway"); err == nil && railwayTag == "signal" {
+				is_signal = true
+			}
+			if idTag, err := OSMUtil.GetTag(*node, "ref"); err == nil && strings.ReplaceAll(idTag, " ", "") == signal.Name[0].Value {
+				has_correct_id = true
+			}
+
+			if is_signal && has_correct_id {
+				insertNewHauptsig(node, kilometrage, *signal, "falling", &notFound)
+			} 
+		}		
+
+		if !found {
+			notFound = append(notFound, signal)
+		} 
+	}
+
+	return notFound	
+}
+
+func processHauptsigS(knoten DBUtil.Spurplanknoten) []*DBUtil.Signal {
+	var notFound = []*DBUtil.Signal{}
+	for _, signal := range knoten.HauptsigS {
+		found := false
+		kilometrage := signal.KnotenTyp.Kilometrierung[0].Value
+		if node_list := anchors[kilometrage]; node_list != nil {
+			for _, node := range node_list {
+				typ, err1 := OSMUtil.GetTag(*node, "type")
+				subtyp, err2 := OSMUtil.GetTag(*node, "subtype")
+				id, err3 := OSMUtil.GetTag(*node, "id")
+				direction, err4 := OSMUtil.GetTag(*node, "direction")
+				if err1 == nil && typ == "element" && err2 == nil && subtyp == "ms" && err3 == nil && id == signal.Name[0].Value && err4 == nil && direction == "rising" {
+					found = true
+					break
+				}
+			}
+		}		
+
+		if found {
+			continue
+		}
+
+		for _, node := range osmData.Node {
+			if len(node.Tag) == 0 {
+				continue
+			}
+
+			is_signal := false 
+			has_correct_id := false
+			if railwayTag, err := OSMUtil.GetTag(*node, "railway"); err == nil && railwayTag == "signal" {
+				is_signal = true
+			}
+			if idTag, err := OSMUtil.GetTag(*node, "ref"); err == nil && strings.ReplaceAll(idTag, " ", "") == signal.Name[0].Value {
+				has_correct_id = true
+			}
+
+			if is_signal && has_correct_id {
+				found = insertNewHauptsig(node, kilometrage, *signal, "rising", &notFound)
+			} 
+		}		
+
+		if !found {
+			notFound = append(notFound, signal)
+		}
 	}
+
+	return notFound	
+}
+
+func searchHauptsigF(knoten DBUtil.Spurplanknoten) {
+	var not_found = []*DBUtil.Signal{} 
+	switch len(anchors) {
+	case 0:
+		fmt.Print("Could not find anchors! \n")
+	case 1:
+		fmt.Print("Could not find enough anchors! \n")
+	default:

```

Comment:
what is the default here? If its more than one, i would suggest a if and early return